### PR TITLE
[FEATURE] Adding Global Error hook to promise mixin

### PR DESCRIPTION
**Pull request by williamsbdev**
This was a desired feature for what I am currently doing at work. It gives me a way to handle certain scenarios (ie 401/403) application wide just by registering my own `onError` function with the `PromiseMixin`. It also allows for normal rejection handling. You will notice one test where I do not handle the rejection and had to monkey patch both the `Ember.Test.adapter.exception` and the `Ember.Logger.error` as there is a `Ember.RSVP.on("error", function(){})` registered for the test build. It will fail the test if you do not handle the error with the rejection function in the `.then()` of your promise. This is good most of the time, but undesirable if your handling via your global hook.

Please let me know if you have any questions/concerns with this PR. Thanks!

I also documented a little bit more how the chaining will work with the new global error hook.

### CODE REVIEW

**Comment by toranb** [comment]
wow dude this was ambitious :) thanks for the great bolt on and above/beyond test coverage as always!

File: tests/dummy/app/routes/onerror-with-multiple-error-handlers.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+var OnErrorRoute = Ember.Route.extend({
+    model: function() {
+        var people = Ember.A();
+        PromiseMixin.xhr("/api/bad-with-handler", "GET").then(function(response) {
+            response.forEach(function(person) {
+                people.pushObject(Ember.Object.create(person));
+            });
+        }, function(response) {
+            ErrorHandler.handled = 1;
+            response.forEach(function(){});
```
Comment (toranb) [inline tests/dummy/app/routes/onerror-with-multiple-error-handlers.js:15]: How did this cause the throw like behavior (forcing the 2nd error handling) ?

Comment (williamsbdev) [inline tests/dummy/app/routes/onerror-with-multiple-error-handlers.js:15]: `response` is not a list so you cannot call `.forEach` which causes the error. We can document that better but I think was a scenario I ran into in a real app.

**Comment by toranb** [comment]
Just did the npm publish/tag for v1.1.0

File: tests/dummy/app/routes/onerror-with-multiple-error-handlers.js (hunk 1 above)
Comment (toranb) [inline tests/dummy/app/routes/onerror-with-multiple-error-handlers.js:15]: Ah that makes sense - did you consider just doing a "threw new Error" instead to be more clear? Would you think this improves the readability for future developers reading the test code?

Comment (williamsbdev) [inline tests/dummy/app/routes/onerror-with-multiple-error-handlers.js:15]: I did consider that and even had that code but for some reason went with
this. I think the `throw new Error` probably makes more sense.

Comment (toranb) [inline tests/dummy/app/routes/onerror-with-multiple-error-handlers.js:15]: Sounds like a plan! Thanks again for all the epic testing you did (learned a ton about chaining) !!!

**Pull request merged**